src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpServer.cs

```diff
@@ -90,10 +91,18 @@ private static bool HandleShutdown(IntPtr pvRequestContext)
             return true;
         }
 
+        private static NativeMethods.REQUEST_NOTIFICATION_STATUS OnAsyncCompletion(IntPtr pvManagedHttpContext, int hr, int bytes)
+        {
+            var context = (HttpProtocol)GCHandle.FromIntPtr(pvManagedHttpContext).Target;
+            context.OnAsyncCompletion(hr, bytes);
+            return NativeMethods.REQUEST_NOTIFICATION_STATUS.RQ_NOTIFICATION_PENDING; // TODO
+        }
+
         private static void CompleteRequest(HttpProtocol context)
         {
             // Post completion after completing the request to resume the state machine
-            context.PostCompletion();
+            //context.PostCompletion();
+            context.IndicateCompletion(NativeMethods.REQUEST_NOTIFICATION_STATUS.RQ_NOTIFICATION_CONTINUE);
```

Comment:
Why?

Reply:
I should talk to @pan-wang about this. Post completion was not causing Continue to be called.

Reply:
We should discuss this. I think we do want to continue on the IIS thread pool not on the .NET Thread pool.

Reply:
@pan-wang agreed that we should be calling indicate completion here.

Reply:
I'm like 89% sure that that's not right.